### PR TITLE
Add AOT build option for arc device

### PR DESCRIPTION
**Pull request by Stonepia**
Cherry-picked from 

https://github.com/intel/torch-xpu-ops/pull/785
https://github.com/intel/torch-xpu-ops/pull/792

- Add AOT build option for arc device.
- Add a message if the user don't need.

### CODE REVIEW

File: cmake/BuildFlags.cmake [hunk 1]
```diff
@@ -115,11 +115,15 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "MSVC"
...
+
   if((DEFINED ENV{TORCH_XPU_ARCH_LIST}) AND NOT ("$ENV{TORCH_XPU_ARCH_LIST}" STREQUAL ""))
     set(SYCL_OFFLINE_COMPILER_AOT_OPTIONS "-device $ENV{TORCH_XPU_ARCH_LIST}")
   else()
-    set(SYCL_OFFLINE_COMPILER_AOT_OPTIONS "-device pvc,xe-lpg")
+    set(SYCL_OFFLINE_COMPILER_AOT_OPTIONS "-device pvc,xe-lpg,ats-m150")
+    message(STATUS "'TORCH_XPU_ARCH_LIST' not set. Using default configuration for AOT build: ${SYCL_OFFLINE_COMPILER_AOT_OPTIONS}." 
+              "Try specifying 'export TORCH_XPU_ARCH_LIST=' from 'pvc,xe-lpg,ats-m150' for current device to accelerate build process.")
   endif()
+
```
Comment (EikanWang) [inline cmake/BuildFlags.cmake:126]:
```suggestion

  set(SYCL_OFFLINE_COMPILER_AOT_OPTIONS "-device ${AOT_TARGETS}")
  message(STATUS "Compile Intel GPU AOT Targets for ${AOT_TARGETS}")
```

File: cmake/BuildFlags.cmake [hunk 2]
```diff
@@ -115,11 +115,15 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "MSVC"
...
   set(SYCL_OFFLINE_COMPILER_CG_OPTIONS "${SYCL_OFFLINE_COMPILER_CG_OPTIONS} -cl-fp32-correctly-rounded-divide-sqrt")
   set(SYCL_OFFLINE_COMPILER_CG_OPTIONS "-options '${SYCL_OFFLINE_COMPILER_CG_OPTIONS}'")
+
   if((DEFINED ENV{TORCH_XPU_ARCH_LIST}) AND NOT ("$ENV{TORCH_XPU_ARCH_LIST}" STREQUAL ""))
     set(SYCL_OFFLINE_COMPILER_AOT_OPTIONS "-device $ENV{TORCH_XPU_ARCH_LIST}")
   else()
-    set(SYCL_OFFLINE_COMPILER_AOT_OPTIONS "-device pvc,xe-lpg")
+    set(SYCL_OFFLINE_COMPILER_AOT_OPTIONS "-device pvc,xe-lpg,ats-m150")
+    message(STATUS "'TORCH_XPU_ARCH_LIST' not set. Using default configuration for AOT build: ${SYCL_OFFLINE_COMPILER_AOT_OPTIONS}." 
+              "Try specifying 'export TORCH_XPU_ARCH_LIST=' from 'pvc,xe-lpg,ats-m150' for current device to accelerate build process.")
```
Comment (EikanWang) [inline cmake/BuildFlags.cmake:124]:
```suggestion
```

File: cmake/BuildFlags.cmake [hunk 3]
```diff
@@ -115,11 +115,15 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "MSVC"
   set(SYCL_OFFLINE_COMPILER_CG_OPTIONS "${SYCL_OFFLINE_COMPILER_CG_OPTIONS} -cl-intel-enable-auto-large-GRF-mode")
   set(SYCL_OFFLINE_COMPILER_CG_OPTIONS "${SYCL_OFFLINE_COMPILER_CG_OPTIONS} -cl-fp32-correctly-rounded-divide-sqrt")
   set(SYCL_OFFLINE_COMPILER_CG_OPTIONS "-options '${SYCL_OFFLINE_COMPILER_CG_OPTIONS}'")
+
   if((DEFINED ENV{TORCH_XPU_ARCH_LIST}) AND NOT ("$ENV{TORCH_XPU_ARCH_LIST}" STREQUAL ""))
     set(SYCL_OFFLINE_COMPILER_AOT_OPTIONS "-device $ENV{TORCH_XPU_ARCH_LIST}")
   else()
```
Comment (EikanWang) [inline cmake/BuildFlags.cmake:121]:
```suggestion
```

**Review by EikanWang — COMMENTED** [review]
In terms of the message, you can refine it.

**Review by EikanWang — APPROVED** [review]

**Comment by EikanWang** [comment]
LGTM. @Stonepia , Please help check if the AOT target setting is as expected.

**Comment by Stonepia** [comment]
> LGTM. @Stonepia , Please help check if the AOT target setting is as expected.
I have checked locally, it should build.

**Comment by dvrogozh** [comment]
What's the binary size increase after adding arc?

**Comment by EikanWang** [comment]
In general, the newly added AOT target increases by about 600~800 megabytes. In terms of the wheel package size,  @chuanqi129 , do you know the accurate binary size?

**Review by EikanWang — APPROVED** [review]

**Comment by chuanqi129** [comment]
> In general, the newly added AOT target increases by about 600~800 megabytes. In terms of the wheel package size, @chuanqi129 , do you know the accurate binary size?

The latest xpu nightly wheel with 3 AOT targets is 750MB, previously is 640MB. You can check the nightly binaries in https://download.pytorch.org/whl/nightly/torch/ directly.
cc @dvrogozh

**Pull request merged**